packages/x-scheduler/src/joy/event-popover/EventPopover.test.tsx

```diff
@@ -77,11 +76,14 @@ describe('<EventPopover />', () => {
     await user.type(screen.getByLabelText(/end date/i), '2025-05-26');
     await user.click(screen.getByRole('button', { name: /save changes/i }));
 
-    const errorDiv = document.querySelector('.EventPopoverDateTimeFieldsError [data-invalid]');
+    const input = screen.getByLabelText(/start date/i);
+    const errorId = input.getAttribute('aria-describedby');
+    expect(errorId).not.to.equal(null);
+    const errorDiv = document.getElementById(errorId!);
     expect(errorDiv?.textContent).to.match(/start.*before.*end/i);
```

Comment:
WDYT about this instead? 🤔 
I prefer avoiding DOM querying whenever possible. 🙈 
P.S. I don't think it's important to ensure null exception prevention in test code. 🙈 
```suggestion
    expect(screen.getDescriptionOf(screen.getByLabelText(/start date/i)).textContent).to.match(
      /start.*before.*end/i,
    );
```